### ISSUE TITLE
[ENHANCEMENT] Evaluation pipeline

### ISSUE THREAD

**Issue by TJ-Solergibert** [swiss-ai/Megatron-LM#2]
Develop a evaluation pipeline which from a checkpoint folder automatically:
- Reshards model checkpoint to run evaluation with the minimum devices (8B --> 1 GPU, 70B --> 2 GPUs)
- Triggers lm-eval harness evaluations
- Logs the result to WANDB

**Comment by ischlag** [comment]
megatron pretrain logging currently lacking tokens per second and weight norms

**Comment by TJ-Solergibert** [comment]
I think we should modify the fork as little as possible to be able to continue syncing from upstream. Clearly seems that `wandb` is not the ideal solution to handle evaluation results. We can start now collecting results in a CSV file storing the most relevant parameters and later we can decide what to do with it.

**Comment by ischlag** [comment]
@AleHD  I don't mind if the logs have a different x-axis as long as it can be converted rather easily. I agree with @TJ-Solergibert that we should change the fork as little as possible but storing to a csv and merging later seems too hacky to me.

**Comment by AleHD** [comment]
Yeah, I don't like the csv solution either. And yes, I wouldn't like to change the code too much, but it looks like adding the OptimizerStep thing will be needed. Otherwise, each `wandb.log(eval)` will create a new Step, so the training logs will need to skip that Step :/

**Comment by ischlag** [comment]
ok, then let's go for it.
